content/docs/guides/crossguard/configuration.md

```diff
@@ -0,0 +1,169 @@
+
+At the top level, the `configSchema` for each Policy has only the `properties` and `required` fields. Within the `properties` field, you may use the entire breadth of [validations keywords provided by the JSON Schema](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.6). Validation keywords are fields like `minLength` and `maxLength` in the below example.
+
+The below example shows a  `ResourceValidationPolicy` that has an optional `message` field, which must be between five and fifty characters. Then, from within the `validateResource` function the configuration value can be accessed using `args.getConfig`. All property fields are optional by default.  
+
+```typescript
+const examplePolicy: ResourceValidationPolicy = {
+    name: "example-policy-with-schema",
+    ...
+    configSchema: {
+        properties: {
+            message: {
+                type: "string",
+                minLength: 5,
+                maxLength: 50,
+            }
+        },
+    },
+    validateResource: validateTypedResource(aws.s3.Bucket, (_, args, reportViolation) => {
+        const config = args.getConfig<{ message?: string }>();
+        if (!config.message) {
+            config.message = "Setting reasonable defaults is recommended!";
+        }
+        reportViolation("Here is the configurable message: " + config.message);
+    }),
+}
+```
+
+The schema in the above example includes an optional property. If the property value is not set, a reasonable message is set from within the `validateResource` function. It's generally better to provide policies that has reasonable defaults and can run without setting any configuration values. Configuration should be used to change behavior or opt-in to more specific checks in the case that a specific configuration property is set. For example, a policy that checks encryption is enabled can be done without any configuration, but an optional configuration property for an "id of the encryption key" can be specified to further enforce that a specific key is being used for the encryption.
+
+#### Required Properties
+
+In some cases, you may need to require a property be set via configuration. This can be done by adding the property to the `required` list as shown in the example below.
+
+```typescript
+const examplePolicy: ResourceValidationPolicy = {
+    name: "example-policy-with-schema",
+    ...
+    configSchema: {
+        properties: {
+            message: {
+                type: "string",
+                minLength: 5,
+                maxLength: 50,
+            }
+        },
+        required: ["message"],
+    },
+    validateResource: validateTypedResource(aws.s3.Bucket, (_, args, reportViolation) => {
+        // We no longer need to add the ? in getConfig function, since the message field is required.
```

Comment:
```suggestion
        // The `?` after the `message` property in the type specified to the `getConfig` function is no
        // longer necessary since `message` is required and will always have a value.
```